src/daemon/daemon.go

```diff
@@ -382,18 +384,24 @@ func (dm *Daemon) Run() error {
 	}
 
 	var err error
+	var elapser = elapse.NewElapser(daemonRunDurationThreshold, logger)
+
 loop:
 	for {
 		select {
+		case <-elapser.Done:
```

Comment:
Do you need this? Can't you can `elapser.CheckForDone` at the beginning of the `for` loop?